doc/changelog/04-tactics/15921-export_micromega.rst

```diff
@@ -0,0 +1,6 @@
+- **Added:**
+  Tactics to obtain micromega witnesses from already reified goals.
+  Using those tactics, the user can develop its own micromega tactics
+  for its own types, using its own parsers
```

Comment:
"reified" is a standard term (at least in PL) meaning, I believe, "expressed in a deeply-embedded way, i.e. as an element of an inductively defined data type of abstract syntax trees".  To "reify" something is to deeply embed it, i.e., to convert it to this representation.

"Witness" is commonly used when there is a big complexity gap between discovering a proof and checking a proof.  The witness is the artifact of discovery that can be checked in much less time than it took to construct.  For example, a witness of the satisfiability of a Boolean formula might be a mapping from free variables to Boolean values.  Such a witness can be checked in linear time, while it may take exponential time to discover.

Reply:
We could add a definition of "reify" somewhere in the doc, though I can't immediately think of a natural place to put it.  But I think we'd need to work to make it more understandable, e.g. we'd need to at least explain embedding.  ("I believe" suggests some uncertainty about the definition.)

Currently, the doc says this about witnesses.  Is that the same or different from your meaning?  Perhaps the following is a bit too limited.

"We say that a type is _inhabited_ if it contains at least one term (i.e. if we can find a term which is associated with this type). We call such terms _witnesses_."

Reply:
> "I believe" suggests some uncertainty about the definition

Yes, I think some might include shallow embeddings in their definition of "reified", I'm not 100% sure.  And mixed embeddings (partially deep but with some leaves allowing shallow embeddings) are certainly included.  So I'm not sure if there's a precise definition, or if it's a cluster concept and my "definition" is the center of the cluster.



> Currently, the doc says this about witnesses. Is that the same or different from your meaning? Perhaps the following is a bit too limited.
> 
> "We say that a type is _inhabited_ if it contains at least one term (i.e. if we can find a term which is associated with this type). We call such terms _witnesses_."

I would call these _witnesses of inhabitation_.  This is indeed the same usage, just more specific.  (It is indeed much easier to check the type of a term than it is to find a term inhabiting an arbitrary type (hence why we have proof checkers and not theorem deciders).)  You can also witness things other than inhabitation, hence "witness" is more general.

Reply:
It appears the witnesses are called `cone expression` in the documentation, I added a link to that term.